compiler/rustc_middle/src/ty/consts/valtree.rs

```diff
@@ -0,0 +1,15 @@
+use super::ScalarInt;
+use rustc_macros::HashStable;
+
+#[derive(Copy, Clone, Debug, Hash, TyEncodable, TyDecodable, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(HashStable)]
+pub enum ValTree<'tcx> {
```

Comment:
Nice and short. :)

But it might be good to have at least a brief doc comment? The plan is for this type to become rather important, after all.

Reply:
For example, the comment should explain that for `enum`s, the *first* child will encode the variant. (Other encodings are possible, e.g. `Branch` could have an `Option<...>` field to encode the variant.)